src/server/dfly_main.cc

```diff
@@ -72,6 +72,7 @@ std::string AbslUnparseFlag(const MaxMemoryFlag& flag) {
 }
 
 ABSL_DECLARE_FLAG(uint32_t, port);
+ABSL_DECLARE_FLAG(bool, random_port);
```

Comment:
i would like to avoid this parameter. 
instead, I suggest that `port == 0` will have random binding and `port < 0` will disable the port.

Reply:
This was requested specifically - https://github.com/dragonflydb/dragonfly/issues/979. Not sure what's best here.

Reply:
@romange `port` is of an unsigned type, so it can't be `< 0`.
However, we could make it of type (signed) int, so that its range is bigger than that of TCP ports, which will allow negative values.
So, yeah, +1 to not adding another flag.. but ideally we won't even have this logic, which is purely needed for testing, but instead assign ports (which we've verified to be unused) in the test. Is that at all possible?

Reply:
@chakaz  unfortunately, this is the only way to do it atomically. @royjacobson wanted to add parallelism to pytests and with parallelism in tests the flow of "first find the port, then bind the port" breaks.

Reply:
Can't we allocate X ports in advance, and hand them out to tests as they go?

Reply:
@royjacobson they requested the way to disable port binding, we do not have to use 0 for that.

Reply:
@chakaz it's technically possible but not straightforward. One needs to bind the ports in python but with special flags that allow sharing. And then dragonfly also should bind them with these flags.  See https://lwn.net/Articles/542629/

Reply:
@romange small ping :)

Reply:
Ok, I accept. last try though - what about setting `-1` as random ? :)

Reply:
switched to `-1` as requested :)